go/cmd/dolt/dolt.go

```diff
@@ -742,6 +752,80 @@ func buildGlobalArgs() *argparser.ArgParser {
 	ap.SupportsString(commands.PrivsFilePathFlag, "", "privilege file", "Path to a file to load and store users and grants. Defaults to `$doltcfg-dir/privileges.db`. Will only be created if there is a change to privileges.")
 	ap.SupportsString(commands.BranchCtrlPathFlag, "", "branch control file", "Path to a file to load and store branch control permissions. Defaults to `$doltcfg-dir/branch_control.db`. Will only be created if there is a change to branch control permissions.")
 	ap.SupportsString(commands.UseDbFlag, "", "database", "The name of the database to use when executing SQL queries. Defaults the database of the root directory, if it exists, and the first alphabetically if not.")
+	ap.SupportsString(commands.ProfileFlag, "", "profile", "The name of the profile to use when executing SQL queries. Defaults to the `default` profile.")
 
 	return ap
 }
+
+func parseGlobalArgsAndProfile(globalConfig config.ReadWriteConfig, args []string) (apr *argparser.ArgParseResults, remaining []string, err error) {
+	apr, remaining, err = globalArgParser.ParseGlobalArgs(args)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	profileName, hasProfile := apr.GetValue(commands.ProfileFlag)
+	profiles, err := globalConfig.GetString(commands.GlobalCfgProfileKey)
+	if err != nil {
+		if err == config.ErrConfigParamNotFound {
+			if hasProfile {
+				return nil, nil, fmt.Errorf("no profiles found")
+			} else {
+				return apr, remaining, nil
+			}
+		} else {
+			return nil, nil, err
+		}
+	}
+	if !hasProfile {
+		defaultProfile := gjson.Get(profiles, "default")
+		if defaultProfile.Exists() {
+			args = append([]string{"--profile", "default"}, args...)
+			apr, remaining, err = globalArgParser.ParseGlobalArgs(args)
+			if err != nil {
+				return nil, nil, err
+			}
+			profileName, hasProfile = apr.GetValue(commands.ProfileFlag)
+		}
+	}
+
+	if hasProfile {
```

Comment:
} else {

Reply:
The logic is a little messy here but I check again for `hasProfile` because I re-parse the args with the `--profile default` flag added. The other option was to copy the code in `hasProfile` after adding the default profile. Which do you think makes more sense?